### DISCUSSION TITLE
Idea about the lambda

### DISCUSSION THREAD

**Discussion by TheSnowfield** [dotnet/csharplang#5910] [Language Ideas] [answered]
We know `()=>"hello"` is a lambda an expr which can be converted to ` Func<string>`,

So why we cannot use these:
```c#
// leads an error :
// Operator '.' cannot be applied to operand of type 'lambda expression'.
(()=>"hello").Invoke();

// leads an error :
// Method, delegate or event is expected
(()=>"hello")(); 
```

Instead of converting them manually:
```c#
// okay
((Func<string>)(()=>"hello")).Invoke();

// okay
((Func<string>)(()=>"hello"))(); 
```

Any ideas? 🤣

**Comment by Richiban** [comment]
I, too, would have expected that to work in C# 10, since lambdas now have a natural type and this works:

```cs
        var f = () => "Hello world";
        
        f();
```

**Reply by TheSnowfield** [reply]
right, you have to do this to bypass these compilation errors, I think it makes smell bad in your code.

**Reply by Xyncgas** [reply]
so we have function pointers in c# now?

**Comment by huoyaoyuan** [comment, marked as answer]
#4748

**Comment by HaloFour** [comment]
Also see: https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-05-10.md#direct-invocation

IMO I hope IIFE never comes to C#.  It exists as a bandaid in Javascript to work around other limitations within the language, like the terrible scoping.  It would be an exceptionally poor replacement for proper [expression blocks](https://github.com/dotnet/csharplang/issues/3086).
